provider/pkg/await/await.go

```diff
@@ -353,14 +355,14 @@ func Read(c ReadConfig) (*unstructured.Unstructured, error) {
 // [3]:
 // https://kubernetes.io/docs/reference/using-api/server-side-apply
 func Update(c UpdateConfig) (*unstructured.Unstructured, error) {
-	client, err := c.ClientSet.ResourceClientForObject(c.Inputs)
+	client, err := c.ClientSet.ResourceClientForObject(c.OldOutputs)
```

Comment:
A subtlety here is that the `APIVersion` would be drawn from old outputs rather than the old inputs.  Is this a concern?  I suppose this line could remain as `c.Input` because it doesn't actually use the object's name.

Reply:
I think that might be a problem in the case where a resource was created with an older APIVersion, the cluster is upgraded to a new version that removes the old APIVersion, and then the program is updated to use the newer APIVersion.

I'm not 100% sure on this, but would lean toward not changing it without further testing along these lines.